### PR TITLE
Source-blacklists updated (sMaSheD miners)

### PR DESCRIPTION
**Pull request by petrmiculek**
added: 
    config file can now specify file_format for each blacklist for easier parsing(JSON, csv, plaintext)
    parsing of JSON blacklists
    crypto-miners Smashed blacklist
removed: 
    discontinued blacklists (ZeuS, urlvir, CryptoIOC, ransomware tracker)

### CODE REVIEW

**Comment by cejkato2** [comment]
Please, check https://github.com/CESNET/Nemea-Modules/blob/master/.travis.yml
To fix the build, it will be (probably) necessary to update python that is used...
Can you make the changes in .travis.yml of this repo?

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in [hunk 1]
```diff
@@ -131,11 +132,26 @@ class Blacklist:
...
-                with suppress(AttributeError):
-                    data = self.cut_csv(data)
+                # JSON
+                try:
+                    data = json.loads(data)  # if not JSON, throws exception
+
+                    data_list = []
+
+                    for i in data:
+                        ip_val = i.get('address')
```
Comment (vaclavbartos) [inline blacklistfilter/blacklist_downloader/bl_downloader.py.in]: Hard-coded key name? That way, it only supports JSON from Smashed, which is not very good design.

The key to look for should be a part of configuration. Whether to parse it as JSON or CSV should be there as well.

**Comment by cejkato2** [comment]
> As a general comment, not related to this PR - why is it configured by such a complicated XML? YAML would be much easier to write and read for users and it's trivial to parse using `pyyaml` package. It shouldn't be difficult to change it.

This is out of scope of this PR. It was due to historical reason, backwards compatibility. There are some deployed instances so we don't want to move from this format now.

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in [hunk 2]
```diff
@@ -131,11 +132,26 @@ class Blacklist:
...
+                        if ip_val not in data_list:
+                            data_list.append(ip_val)
+
+                    new_entities = sorted(data_list)  # , key=type(self).comparator
```
[the remark this answers is not in the document]
Comment (petrmiculek) [reply blacklistfilter/blacklist_downloader/bl_downloader.py.in]: Now uses a set, thanks for the advice.

File: blacklistfilter/blacklist_downloader/bl_downloader_config.xml [hunk 3]
```diff
@@ -186,51 +158,30 @@
...
-                <element name="name">Malware Domains</element>
+                <element name="name">PhishTank</element>
                 <!-- Address from which the blacklist will be downloaded -->
-                <element name="source">http://mirror1.malwaredomains.com/files/justdomains</element>
+                <element name="source">http://data.phishtank.com/data/online-valid.csv</element>
                 <!--Category of the blacklist, it SHOULD match some of the IDEA categories (idea.cesnet.cz)-->
-                <element name="category">Malware</element>
+                <element name="category">Fraud.Phishing</element>
                 <!-- Download interval in minutes -->
                 <element name="download_interval">10</element>
```
[the remark this answers is not in the document]
Comment (petrmiculek) [reply blacklistfilter/blacklist_downloader/bl_downloader_config.xml]: Good idea, I've made the PhishTank download interval longer. I haven't reviewed other lists' intervals yet, though. Will do.

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in (hunk 1 above)
Comment (petrmiculek) [reply blacklistfilter/blacklist_downloader/bl_downloader.py.in]: Config file has been expanded to include these attributes.

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in [hunk 4]
```diff
@@ -124,18 +140,38 @@ class Blacklist:
...
                 data = req.content
 
-                # If CSV, cut the desired column
-                with suppress(AttributeError):
-                    data = self.cut_csv(data)
+                # JSON
+
+                if self.file_format == 'JSON':
+                    try:
+                        data = json.loads(data)  # if not JSON, raises ValueError or TypeError how to catch both?
```
Comment (vaclavbartos) [inline blacklistfilter/blacklist_downloader/bl_downloader.py.in]: This catches both exceptions:
`except (ValueError, TypeError) as e:`

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in [hunk 5]
```diff
@@ -147,12 +183,12 @@ class Blacklist:
 
                     logger.info('Updated {}: {}'.format(type(self).__name__, self.name))
 
-                if not new_entities:
+                if new_entities is None:
                     logger.warning('{}, {}: No valid entities found'.format(type(self).__name__, self.name))
```
Comment (vaclavbartos) [inline blacklistfilter/blacklist_downloader/bl_downloader.py.in]: The preivous version (`not new_entities`) was actually better, as it matches `None` as well as empty list/set - which is what we want to warn about.

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in (hunk 4 above)
Comment (petrmiculek) [reply blacklistfilter/blacklist_downloader/bl_downloader.py.in]: Done

File: blacklistfilter/blacklist_downloader/bl_downloader.py.in (hunk 5 above)
Comment (petrmiculek) [reply blacklistfilter/blacklist_downloader/bl_downloader.py.in]: Fixed

**Comment by cejkato2** [comment]
Thank you for the modifications.

**Pull request merged**